### PR TITLE
external_services: automatically add authorization for GitLab and GitHub on Cloud

### PR DESCRIPTION
**Pull request by unknwon**
This PR automatically adds `"authorization"` field to user code host connections on dotcom for GitLab.com and GitHub.com at creation to enforce repository permissions.

The reason I don't do this inside `ValidateConfig` is because I think "validation" shouldn't involve mutating the config.

I'm gonna fix and add corresponding tests if people agree with the current approach.

### TODO

- [x] Tests

### CODE REVIEW

File: internal/db/external_services.go [hunk 1]
```diff
@@ -217,41 +217,35 @@ func (e *ExternalServiceStore) ValidateConfig(ctx context.Context, opt ValidateE
 	switch opt.Kind {
 	case extsvc.KindGitHub:
 		var c schema.GitHubConnection
-		if err = json.Unmarshal(normalized, &c); err != nil {
+		if err = jsoniter.Unmarshal(normalized, &c); err != nil {
```
Comment (unknwon) [inline internal/db/external_services.go:220]: `jsoniter` has been used in code-intel and I don't see a reason why we should continue using the stdlib `encoding/json` here.

Comment (tsenart) [reply internal/db/external_services.go:220]: Does it support JSON with comments?

**Review by tsenart — APPROVED** [review]
LGTM, but could it be safer to change authorization code to always instantiate an authorization provider for user added external services, regardless of the config?

**Review by ryanslade — APPROVED** [review]

File: internal/db/external_services.go (hunk 1 above)
Comment (unknwon) [reply internal/db/external_services.go:220]: No, it does not unfortunately.

**Comment by unknwon** [comment]
> but could it be safer to change authorization code to always instantiate an authorization provider for user added external services, regardless of the config?

@tsenart I think that would make our authz logic dependent on application-level logic, we need to persist states in DB so pure-SQL authz can continue to work.

File: internal/db/external_services.go (hunk 1 above)
Comment (tsenart) [reply internal/db/external_services.go:220]: Then won't it break? Our external service configs have comments.

Comment (unknwon) [reply internal/db/external_services.go:220]: The variable `normalized` is clean/standard JSON at this point :D

**Pull request merged**